src/arns.lua

```diff
@@ -512,7 +524,7 @@ function arns.assertValidBuyRecord(name, years, purchaseType, processId)
 
 	-- assert processId is valid pattern
 	assert(type(processId) == "string", "Process id is required and must be a string.")
-	assert(utils.isValidAOAddress(processId), "Process Id must be a valid AO signer address..")
+	assert(utils.isValidAddress(processId, true), "Process Id must be a valid address.")
```

Comment:
I think this might be one where we want to force users to specify allowing unsafe because if they get this wrong, they can brick the name. e.g. they wouldn't be able to use the ANT to reassign to another process ID.

Reply:
f602db8